OctoAwesome/OctoAwesome.Basics/ComplexPlanetGenerator.cs

```diff
@@ -21,29 +21,20 @@ public IPlanet GeneratePlanet(Guid universe, int id, int seed)
 
         public IChunkColumn GenerateColumn(IDefinitionManager definitionManager, IPlanet planet, Index2 index)
         {
-            IDefinition[] definitions = definitionManager.GetDefinitions().ToArray();
-            //TODO More Generic, überdenken der Planetgeneration im allgemeinen (Heapmap + Highmap + Biome + Modding)
-            IBlockDefinition sandDefinition = definitions.OfType<SandBlockDefinition>().FirstOrDefault();
-            ushort sandIndex = (ushort)(Array.IndexOf(definitions.ToArray(), sandDefinition) + 1);
-
-            IBlockDefinition snowDefinition = definitions.OfType<SnowBlockDefinition>().FirstOrDefault();
-            ushort snowIndex = (ushort)(Array.IndexOf(definitions.ToArray(), snowDefinition) + 1);
-
-            IBlockDefinition groundDefinition = definitions.OfType<GroundBlockDefinition>().FirstOrDefault();
-            ushort groundIndex = (ushort)(Array.IndexOf(definitions.ToArray(), groundDefinition) + 1);
-
-            IBlockDefinition stoneDefinition = definitions.OfType<StoneBlockDefinition>().FirstOrDefault();
-            ushort stoneIndex = (ushort)(Array.IndexOf(definitions.ToArray(), stoneDefinition) + 1);
-
-            IBlockDefinition waterDefinition = definitions.OfType<WaterBlockDefinition>().FirstOrDefault();
-            ushort waterIndex = (ushort)(Array.IndexOf(definitions.ToArray(), waterDefinition) + 1);
-
-            IBlockDefinition grassDefinition = definitions.OfType<GrassBlockDefinition>().FirstOrDefault();
-            ushort grassIndex = (ushort)(Array.IndexOf(definitions.ToArray(), grassDefinition) + 1);
-
             if (!(planet is ComplexPlanet))
                 throw new ArgumentException("planet is not a Type of ComplexPlanet");
 
+            IDefinition[] definitions = definitionManager.GetDefinitions().ToArray();
+            //TODO More Generic, überdenken der Planetgeneration im allgemeinen (Heapmap + Highmap + Biome + Modding)
+            // +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+            ushort sandIndex = definitionManager.GetDefinitionIndexByName(Languages.OctoBasics.Sand);
```

Comment:
Das ist meiner Meinung nach nicht so gut, den lokalisieten Namen zu verwenden. Verleitet einen ENtwickler, der nur in Sprache X testet deren hardcoded Wert zu verwenden, und in Sprache Y gehts dann nicht mehr. rweiterungsübergreifend ist auch per se kein gegenseitiger Zugriff auf die Lokalisierung möglich.

**Wichtig**: Diese A´nmerkung gilt gleich für alle anderen Stellen, an denen dies so verwendet wird.

Reply:
Ja das ist mir bewusst. Ich plädiere für ein festes Token auch Server übergreifend, Aber nebenbei gesagt das ist Modder Code, wer kann den Modder schon zwingen in verschiedenen sprachen zu Coden :D
aber hast recht! das wäre ohnehin nur vorübergehend so geblieben